components/elements/AccountDropdown.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import styled from 'styled-components';
+import useUser from '../auth/User';
+
+const AccountDropdown = ({ children }) => {
+  const user = useUser();
+
+  let userInitial;
+  if (user && user.name) {
+    userInitial = user.name[0].toUpperCase();
```

Comment:
I see setting uppercase more as a cosmetic thing instead of a logical issue. I would prefer you do that with css and keep the data as clean as possible.
The advantage is that you can also have a nice destructured code:
```javascript
  const [ name ] = user;
```

styled component/css

```javascript

const userInitial = styled.section`
 /// other css
text-transform: uppercase
`;